src/core/algorithms/gfd/parser.cpp

```diff
@@ -0,0 +1,287 @@
+#include <boost/algorithm/string.hpp>
+#include <boost/bind/bind.hpp>
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/graphviz.hpp>
+#include <boost/property_map/function_property_map.hpp>
+// #include <boost/config.hpp>
+// #include <boost/regex.hpp>
+// #include <libs/graph/src/read_graphviz_new.cpp>
+// #include "/usr/local/lib/libboost_graph.so"
+
+// #include "/home/anton/boost_1_72_0/libs/graph/src/read_graphviz_new.cpp"
+// #include "/home/anton/boost_1_72_0/libs/graph/src/read_graphviz_new.cpp"
+// #include "/usr/local/include/boost/graph/detail/read_graphviz_new.cpp"
+
+#include "parser.h"
+
+namespace GFDtools {
+
+typedef boost::property_map<graph_t, std::map<std::string, std::string> vertex::*>::type AMap;
+typedef boost::property_map<graph_t, std::string edge::*>::type RMap;
+
+namespace {
+template <typename AMap>
+struct NewAttr {
+    typedef boost::shared_ptr<boost::dynamic_property_map> Ptr;
+
+private:
+    template <typename PMap>
+    static Ptr make_dyn(PMap m) {

```

Comment:
We use `PascalCase` for method names --- make sure to rename all the other methods in `gfd`